src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexRunner.cs

```diff
@@ -136,6 +136,32 @@ public abstract class RegexRunner
             runtextbeg = textbeg;
             runtextend = textend;
 
+            if (!regex.RightToLeft && regex.aho != null)
```

Comment:
This is the crux of the prototype. Returns on the first match for now, but can easily be extended to work for multiple matches.

Reply:
Why for Boyer Moore do we implement it explicitly in the emitted code (RegexCompiler) but for AC we can get away with doing all the job in this utility function? Is it efficiency vs. code complexity? (I haven't thought this all through yet)

Reply:
Actually I've only prototyped the interpreted paths. I have no idea how the compiled paths will need to turn out